### PR TITLE
Added provisioning of libvirt and controller nodes

### PR DESCRIPTION
**Pull request by TorLdre**
A very redimentary, non secure implementation of libvirt. This will suffice for our needs in the short term. Later on, security should be added. This implementation allows everyone to connect to libvirt and manage VMs and resources, as well as connection to VNC console to each VM.

### CODE REVIEW

File: hieradata/defaults/controller.yaml [hunk 1]
```diff
@@ -1,7 +1,63 @@
...
+libvirt::qemu_vnc_listen:         '0.0.0.0'
+libvirt::qemu_vnc_sasl:           true
+libvirt::qemu_vnc_tls:            true
+
+libvirt_networks:
+  direct-net:
+    ensure:              enabled
+    autostart:           true
+    forward_mode:        bridge
+    bridge:              br2
```
Comment (beddari) [inline hieradata/defaults/controller.yaml]: br0?

**Comment by TorLdre** [comment]
beddari. Yes, should be br0. Would work with br2 though. I'll fix.

**Comment by TorLdre** [comment]
Remnants from testing :)

**Comment by beddari** [comment]
Rebased and merged as #75

**Pull request closed without merging**